pkgs/applications/virtualization/qemu/default.nix

```diff
@@ -233,13 +235,21 @@ stdenv.mkDerivation rec {
     mkdir -p $ga/bin
     cp $out/bin/qemu-ga $ga/bin/
     remove-references-to -t $out $ga/bin/qemu-ga
+  '' + lib.optionalString stdenv.isLinux ''
+    # copy qemu-plugins (TCG plugins) to separate output
+    mkdir -p $plugins/libexec/qemu
+    for f in contrib/plugins/*.so; do
+      cp $f $plugins/libexec/qemu/
+    done
   '' + lib.optionalString gtkSupport ''
     # wrap GTK Binaries
     for f in $out/bin/qemu-system-*; do
       wrapGApp $f
     done
   '';
   preBuild = "cd build";
+  # contrib/plugins/Makefile requires support for -soname option
+  postBuild = lib.optionals stdenv.isLinux "make plugins";
```

Comment:
Considering `make plugins` just runs `make` in the plugins directory, and the plugins directory says "This Makefile example is fairly independent from the main makefile", I wonder if it'd be better to skip the top-level Makefile and do `make -C plugins` here?  Also remember to pass the standard make flags variables. (From memory: `$makeFlags "${makeFlagsArray[@]}" $buildFlags "${buildFlagsArray[@]}"`, but do check.)